### PR TITLE
Matches toBe(null) and toEqual(null) in addition to toBeNull() in prefer-document

### LINKED ISSUE #152: match `toBe(null)` and possibly `toEqual(null)` in addition to `toBeNull`
In a few rules (eg: `prefer-document`) `toBeNull()` will be replaced by `toBeInTheDocument()` which is nice.

Is it possible to match `toBe(null)` and `toEqual(null)` as well?

Thanks!

### PR DESCRIPTION
**Pull request by julienw**
Fixes #152

**Checklist**:

- [X] Documentation
- [X] Tests
- [X] Ready to be merged

### CODE REVIEW

File: src/rules/prefer-in-document.js [hunk 1]
```diff
@@ -24,16 +24,22 @@ export const meta = {
...
+    matcherArguments[0].value === null;
+}
+
 function usesToHaveLengthZero(matcherNode, matcherArguments) {
   return matcherNode.name === "toHaveLength" && matcherArguments[0].value === 0;
 }
 
 export const create = (context) => {
-  const alternativeMatchers = /(toHaveLength|toBeDefined|toBeNull)/;
+  const alternativeMatchers = /^(toHaveLength|toBeDefined|toBeNull|toBe|toEqual)$/;
```
Comment (julienw) [inline src/rules/prefer-in-document.js:42]: Without the added `^` and `$`, the `toBe` part matches `toBeInTheDocument` 😱

**Comment by julienw** [comment]
(looking at the failures right now)

**Comment by benmonro** [comment]
@allcontributors please add @julienw for code and tests

**Comment by julienw** [comment]
```
[lint]   61:3  error  Function 'check' has a complexity of 15. Maximum allowed is 14  complexity
```
The part I added is so small 😓
Is it worth trying to fix this?

**Comment by benmonro** [comment]
@julienw if you can easily fix it that would be great, if not you can eslint disable for now, and please add an issue to resolve it.

many thanks

**Comment by julienw** [comment]
BTW I can't run eslint locally, I get this error:
```
Error: package.json » ./node_modules/kcd-scripts/eslint.js » /home/julien/travail/git/eslint-plugin-jest-dom/node_modules/kcd-scripts/node_modules/eslint-config-kentcdodds/jest.js:
	Environment key "jest/globals" is unknown
```
Not sure why :-)

**Comment by benmonro** [comment]
ugh, kcd scripts has been a bit problematic for me... one of these days i'll remove it.

File: src/rules/prefer-in-document.js [hunk 2]
```diff
@@ -79,6 +85,14 @@ export const create = (context) => {
       }
     }
 
+    // toBe() or toEqual() are only invalid with null
+    if (
+      (matcherNode.name === "toBe" || matcherNode.name === "toEqual") &&
+      !usesToBeOrToEqualWithNull(matcherNode, matcherArguments)
+    ) {
```
Comment (julienw) [inline src/rules/prefer-in-document.js:92]: I merged the 2 if in one condition. I believe this is _less_ readable, not more, but well let's see if eslint is happier with this.

File: src/rules/prefer-in-document.js [hunk 3]
```diff
@@ -3,6 +3,8 @@
  * @author Anton Niklasson
  */
 
+/*eslint complexity: ["error", {"max": 20}]*/
```
Comment (julienw) [inline src/rules/prefer-in-document.js:6]: This was the easiest way to fix the eslint issue. I believe the problem is in other parts of the code than the one I added.

Also `20` is the default in eslint, it looks like some other other preset this project is using set it to 15, so I think this is reasonable to put it back to `20` here.

Tell me what you think!

**Comment by julienw** [comment]
This is now all green :-)

**Review by benmonro — APPROVED** [review]

**Comment by benmonro** [comment]
Thanks again @julienw BTW, keep an eye out on the smoke test job, it's a pretty sweet tool that runs this plugin against a ton of projects as a way of preemptively detecting possible bugs in the lint code.  https://github.com/testing-library/eslint-plugin-jest-dom/actions/workflows/smoke-test.yml. now that your code is in there, if there's any potential use cases that might cause an unhandled exception, we can fix it before it gets reported. :)

**Comment by julienw** [comment]
:+1: 
I'm sure you'll ping me anyway ;-)

BTW it's likely we get new (fixable) failures in existing projects, so this will be in a minor or major release, I guess?

**Pull request merged**